### DISCUSSION TITLE
shift pageup and shift pagedown

### DISCUSSION THREAD

**Discussion by brianmay** [zellij-org/zellij#2934] [General]
When I push shift pageup or shift pagedown, I instinctively expect to able to page up or down through the scroll buffer. Because this it what some terminals support.

Instead it changes the layout of the window. Which I really don't want.

But if I look through the default keyboard bindings, I don't see anything like this. And If I try to add a keyboard binding for "Ctrl PageUp" or "Shift PageUp" it says "Invalid Key".

**Comment by cristiand391** [comment]
> When I push shift pageup or shift pagedown, I instinctively expect to able to page up or down through the scroll buffer. Because this it what some terminals support.
Instead it changes the layout of the window. Which I really don't want.

Seems like a bug:
https://github.com/zellij-org/zellij/issues/2441

Also, not sure `Shift` works as a modifier (just see `Ctrl` and `Alt` as modifiers here: https://zellij.dev/documentation/keybindings-keys).


So, you'll need to use another key combination for [HalfPageScrollDown](https://zellij.dev/documentation/keybindings-possible-actions#halfpagescrolldown) and [HalfPageScrollDown](https://zellij.dev/documentation/keybindings-possible-actions#halfpagescrolldown).

**Comment by brianmay** [comment]
Thanks for that. Also found #2673 which looks like an earlier report of the same thing.

I don't mind not being able to bind shift+pageup and shift+pagedown, but I really would like to disable the built in binding, because I am accidentally messing up my layout every time I want to scroll up.

**Reply by cristiand391** [reply]
You can unbind it in your config, the default binding for that is in the search mode so you should do something like:

```kdl
search {
  unbind "Ctrl f" "PageDown" "Right" "l" { PageScrollDown; }
  unbind "Ctrl b" "PageUp" "Left" "h" { PageScrollUp; }
}

```

docs:
https://zellij.dev/documentation/keybindings-binding

see also:
https://github.com/zellij-org/zellij/discussions/3026#discussioncomment-7959856

**Reply by ghost** [reply]
I don't think it is that simple, because there is nothing binding on "PageUp" or "PageDown" that I can unbind.

Tested, and it didn't appear to work.

The binding is for "Ctrl-PageUp" or "Shift-PageUp", but as mentioned above, I can't specify that in the unbind.

**Comment by Vinh-CHUC** [comment]
+1 specifically for the inability to use "Shift" as a modifier

**Comment by adsick** [comment]
Is it still valid? I was able to add this to my config and it seems to work fine:
```kdl
        bind "Shift PageDown" { PageScrollDown; }
        bind "Shift PageUp" { PageScrollUp; }
```
I added it to already existing
```kdl
shared_except "locked" "scroll" "search" 
```
section.

Would be nice to have it by default by the way.

**Reply by brianmay** [reply]
This may have fixed. I haven't tested it extensively. https://github.com/zellij-org/zellij/issues/2441#issuecomment-2472835103

**Reply by brianmay** [reply]
Tested, appears to be working fine for me.
